src/Expectation.php

```diff
@@ -65,6 +65,20 @@ public function __construct(
         // ..
     }
 
+    /**
+     * Allows you to target a value in a table
+     *
+     * @return self<TAndValue>
+     */
+    public function at(string|int $key): Expectation
```

Comment:
If this is to be implemented, I think an option for dot-notation syntax would be much nicer.

```php
->at('0.1.2')
```

Personally, I'm not sure of the benefit of this, can higher order access not be used?

```php
->{1}->{2}
->name->firstName
```

Reply:
> If this is to be implemented, I think an option for dot-notation syntax would be much nicer.

That's a great idea, I hadn't thought of that. I'm looking at reworking my PR in that direction this week. 

> Personally, I'm not sure of the benefit of this, can higher order access not be used?


Yes, it's usable but the function I want to implement is designed to add readability too. On a structure that's not too complex you can use higher order access but I think that on a more complex structure the risk of getting lost in the reading is greater.

I'd also like to find a way of being able to chain my function without having to use `and()` to provide even more readability. But at the moment I haven't figured out how I could do that.